cmake-tool/helpers/cpio.cmake

```diff
@@ -44,9 +44,15 @@ function(MakeCPIO output_name input_files)
         set(archive_symbol ${MAKE_CPIO_CPIO_SYMBOL})
     endif()
 
+    get_filename_component(cpio_lib_name "${output_name}" NAME)
+    get_filename_component(cpio_lib_dir "${output_name}" DIRECTORY)
+    if(NOT cpio_lib_dir)
+        set(cpio_lib_dir "${CMAKE_CURRENT_BINARY_DIR}")
+    endif()
+
     # intermediate files
-    set(cpio_archive "archive.${output_name}.cpio")
-    set(cpio_archive_s "${output_name}.S")
+    set(cpio_archive "${cpio_lib_dir}/${cpio_lib_name}.cpio")
+    set(cpio_archive_s "${cpio_lib_dir}/${cpio_lib_name}.S")
```

Comment:
What happens to existing calls to MakeCPIO that don't pass in any DIRECTORY path component?
Eg in the elfloader: `MakeCPIO(archive.o "${cpio_files}" CPIO_SYMBOL _archive_start)`, `cpio_lib_dir` would be set to "" and then the cpio_archive would be created at `/${cpio_lib_name}.cpio` I think?

Reply:
I will check this case again. I thought I caught all corner cases from testing the practical usage, as our internal CI runs passed.

Reply:
That case is handeled, there is a check in line 49 that handles this:
```
    if(NOT cpio_lib_dir)
        set(cpio_lib_dir "${CMAKE_CURRENT_BINARY_DIR}")
    endif()
```

Reply:
Ok thanks